mountpoint-s3-fs/src/superblock.rs

```diff
@@ -321,6 +323,91 @@ impl<OC: ObjectClient + Send + Sync> Superblock<OC> {
         trace!("Added handle with id: {}", handle_id);
         Ok(handle_id)
     }
+
+    async fn start_reading(&self, looked_up_inode: LookedUpInode, fh: u64) -> Result<Lookup, InodeError> {
+        let pending_upload_hook = {
+            let mut locked_inode = looked_up_inode.inode.get_mut_inode_state()?;
+            if matches!(locked_inode.write_status, WriteStatus::LocalUnopened)
+                || !self.inner.open_handles.try_add_reader(&locked_inode, fh)
+            {
+                return Err(InodeError::InodeNotReadableWhileWriting(looked_up_inode.inode.err()));
+            }
+            if matches!(locked_inode.write_status, WriteStatus::LocalOpenForWriting) {
+                locked_inode.write_status = WriteStatus::Remote;
```

Comment:
nit: comment / debug log about what's happening here: open for read "overtook" the write handle so switching status to `WriteStatus::Remote`